platform/android/src/style/sources/geojson_source.cpp

```diff
@@ -16,7 +16,13 @@
 #include "../conversion/url_or_tileset.hpp"
 
 #include <string>
+#include <mbgl/util/shared_thread_pool.hpp>
 
+// GeoJSONSource uses a "coalescing" model for high frequency asynchronous data update calls,
+// which in practice means, that any update that started processing is going to finish
+// and the last scheduled update is going to finish as well. Any updates scheduled during processing can be canceled.
+// Conversion from Java features to core ones is done on a worker thread and once finished,
+// the control of the updated is passed to the core on the main thread.
```

Comment:
I had a little trouble reading this on the first pass

> "the control of the updated is passed to the core on the main thread"

Maybe that could be something like

> " ownership of the converted features is returned to the calling thread"

?